### PR TITLE
Feature/add fly form

### LINKED ISSUE #1: Add New Flies - UI
As a user, I can click an add button to add a new fly to my fly box.

- [x] [Button to Add Fly Page](#15)
- [x] [Add Fly Input Fields](#14)
- [x] [Clear Input Fields Button](#13)
- [x] [Submit New Fly Button](#12)

### LINKED ISSUE #12: Submit New Fly Button
There should be a button to submit a new fly where the user's flies database is updated and redirected to the main page.

### LINKED ISSUE #13: Clear Input Field Button
There should be a clear button to clear all input fields and return to home page

### PR DESCRIPTION
**Pull request by TimNguyen21**
### Issue(s) related to this PR:
* closes #1 
* closes #12 
* closes #13
 
 
 ### Type of change made:
* [ ] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] Styling- no new features

#### If this submission is a new feature:
* [x] Have you added an explanation of what your changes do?
* [x] Have you lint your code locally prior to submission?

### Change Log:
* ApiCalls.js - Created fetch POST call to add a fly to database
* actions/index.js - Added action creators for adding fly, updating fly, and clear add fly input fields
* components/FlyForm.js - Modified the correct labels to match BE database
* containers/AddFly.js - Added fetching properties to update BE database and Redux Store
* reducers/currentFlies.js - Created current fly state to be used for Redux Store
* reducers/currentFlyEntry.js - Created cases for Current Fly Entry
* reducers/index.js - Added store for Current Fly entry in Redux Store

### Were there any challenges that arose? If so, how were the addressed?
* Fetching in the Add Fly Form was giving an error, but the BE team had fixed the problem and there were no issues after the fix.

### Technical Debt:
* n/a

### Testing:
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?
* [ ] All new and existing tests passed?
* [ ] Have you run linter?

**Test coverage** = 0

### CODE REVIEW

File: ApiCalls.js [hunk 1]
```diff
@@ -9,3 +9,19 @@ export const fetchFlies = async () => {
     console.error(error);
   }
 };
+
+export const addFlyToAPI = async (flyData) => {
```
Comment (rachael-t) [inline ApiCalls.js:13]: Great job with this post request!

File: containers/AddFly.js [hunk 2]
```diff
@@ -1,25 +1,50 @@
...
-      <View style={{flex: 1, flexDirection: 'row', justifyContent: 'center'}}>
-        <TouchableOpacity style={styles.button}
-          onPress = {() => {alert('Go to Home Page')}}>
-          <Text style={styles.button}>Cancel</Text>
-        </TouchableOpacity>
+  addFly = () => {
+    const currentFlyEntry = this.props.currentFlyEntry
+    addFlyToAPI(currentFlyEntry)
+      .then(data => {
+          this.props.addFly(data.data);
```
Comment (rachael-t) [inline containers/AddFly.js]: Great job! So whenever a fly is successfully posted to the API via `addFlyToAPI`, we are returned that new fly's data object which we then store to state via the `addFly` action creator, correct?

Comment (TimNguyen21) [reply containers/AddFly.js]: Yes. If it is a successful request, the response of the successful POST request returns an object of the Fly that is submitted.  So, that response is then added to the FliesData in the Redux store.

Comment (TimNguyen21) [reply containers/AddFly.js]: I think we need to make a conditional where if there is a bad response, then the response shouldn't be added to the FliesData.  Let's keep that in mind when we make our future updates. I believe one way to take on that problem is by adding a disable feature to the button where the submit button is disabled until all fields are entered in the Add Fly Form.

Comment (rachael-t) [reply containers/AddFly.js]: Sounds good - I think that's a great idea.

**Pull request merged**